metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorPartitionService.java

```diff
@@ -0,0 +1,120 @@
+                tableInfo,
+                context,
+                partitionsRequest.getFilter(),
+                partitionsRequest.getPartitionNames(),
+                partitionsRequest.getSort()
+            ),
+            partitionsRequest.getPageable()
+        );
+    }
+
+    /**
+     * {@inheritDoc}.
+     */
+    @Override
+    public List<String> getPartitionKeys(@NonNull final ConnectorRequestContext requestContext,
+                                         @NonNull final QualifiedName tableName,
+                                         @NonNull final PartitionListRequest partitionsRequest,
+                                         @NonNull final TableInfo tableInfo) {
+        return getPartitions(requestContext, tableName, partitionsRequest, tableInfo).stream()
+                   .map(info -> info.getName().getPartitionName())
+                   .collect(Collectors.toList());
+    }
+
+    /**
+     * {@inheritDoc}.
+     */
+    @Override
+    public int getPartitionCount(@NonNull final ConnectorRequestContext requestContext,
+                                 @NonNull final QualifiedName table,
+                                 @NonNull final TableInfo tableInfo) {
+        return icebergTableHandler.getPartitions(
+            tableInfo,
+            context,
+            null, // filer expression
+            null, // partition ids
+            null // sort
+        ).size();
+    }
+
+    /**
+     * {@inheritDoc}.
+     */
+    @Override
+    public List<String> getPartitionUris(@NonNull final ConnectorRequestContext requestContext,
+                                         @NonNull final QualifiedName table,
+                                         @NonNull final PartitionListRequest partitionsRequest,
+                                         @NonNull final TableInfo tableInfo) {
+        return getPartitions(requestContext, table, partitionsRequest, tableInfo).stream()
+                   .map(partitionInfo -> partitionInfo.getSerde().getUri())
+                   .collect(Collectors.toList());
+    }
+
+    /**
+     * {@inheritDoc}.
+     */
+    @Override
+    public PartitionInfo get(@NonNull final ConnectorRequestContext requestContext,
+                             @NonNull final QualifiedName partitionName) {
+        final QualifiedName tableName = QualifiedName.ofTable(
+            partitionName.getCatalogName(),
+            partitionName.getDatabaseName(),
+            partitionName.getTableName()
+        );
+
+        final TableInfo tableInfo = tableService.get(requestContext, tableName);
+
+        final List<PartitionInfo> partitions = icebergTableHandler.getPartitions(
+            tableInfo,
+            context,
+            null,
+            Collections.singletonList(partitionName.getPartitionName()),
+            null
+        );
+
+        return partitions.stream()
+                   .filter(partitionInfo -> partitionInfo.getName().equals(partitionName))
```

Comment:
Seems it should be safe without this filter since the name filter was already passed in the earlier call, though checking again here is also fine.

Reply:
Good eye; does not harm except for the extra allocations and checks. Will remove it next time I make a change in this repo.